sdk/core/core/src/az_log.c

```diff
@@ -176,6 +176,28 @@ static az_result _az_log_http_response_msg(
   return _az_log_http_request_msg(log_msg_bldr, hrb, 1);
 }
 
+static az_result _az_log_http_retry_msg(
+    az_span_builder * const log_msg_bldr,
+    int16_t const attempt,
+    int32_t const delay_msec) {
+  AZ_RETURN_IF_FAILED(az_span_builder_append(log_msg_bldr, AZ_STR("HTTP Retry attempt")));
+
+  if (attempt >= 0) {
+    AZ_RETURN_IF_FAILED(az_span_builder_append(log_msg_bldr, AZ_STR(" #")));
+    AZ_RETURN_IF_FAILED(az_span_builder_append_uint64(log_msg_bldr, (uint64_t)attempt));
+  }
+
+  AZ_RETURN_IF_FAILED(az_span_builder_append(log_msg_bldr, AZ_STR(" will be made")));
+
+  if (delay_msec >= 0) {
```

Comment:
Take this if out. If we're logging retry logic then let's see the 0msec log entry too.